services/typescript-service.ts

```diff
@@ -0,0 +1,316 @@
+					typeScriptDefinitionsFiles = this.getDefaultTypeScriptDefinitionsFiles(options.pathToDefaultDefinitionFiles).wait();
+				}
+
+				this.$fs.writeFile(typeScriptCommandsFilePath, this.typeScriptFiles.concat(typeScriptDefinitionsFiles).concat(typeScriptCompilerOptions).join(" ")).wait();
+
+				let typeScriptCompilerSettings = this.getTypeScriptCompiler().wait();
+
+				// Log some messages
+				this.$logger.out("Compiling...".yellow);
+				_.each(this.typeScriptFiles, file => {
+					this.$logger.out(`### Compile ${file}`.cyan);
+				});
+				this.$logger.out(`Using tsc version ${typeScriptCompilerSettings.version}`.cyan);
+
+				// Core compilation
+				return this.runTranspilation(projectDir, typeScriptCompilerSettings.pathToCompiler, { typeScriptCommandsFilePath }).wait();
+			} else {
+				return this.transpileWithDefaultOptions(projectDir, compilerOptions).wait();
+			}
+		}).future<string>()();
+	}
+
+	// Uses tsconfig.json if it exists
+	public transpileWithDefaultOptions(projectDir: string, options?: ITypeScriptTranspileOptions): IFuture<string> {
+		return ((): string => {
+			options = options || {};
+			let compilerOptions = this.getCompilerOptions(projectDir, options).wait();
+			this.noEmitOnError = compilerOptions.noEmitOnError;
+			let typeScriptCompilerSettings = this.getTypeScriptCompiler().wait();
+			this.$logger.out(`Using tsc version ${typeScriptCompilerSettings.version}`.cyan);
+
+			// Core compilation
+			return this.runTranspilation(projectDir, typeScriptCompilerSettings.pathToCompiler, { compilerOptions }).wait();
+		}).future<string>()();
+	}
+
+	public getTypeScriptFiles(projectDir: string): IFuture<ITypeScriptFiles> {
+		return ((): ITypeScriptFiles => {
+			// Skip root's node_modules
+			let rootNodeModules = path.join(projectDir, "node_modules");
+			let projectFiles = this.$fs.enumerateFilesInDirectorySync(projectDir,
+				(fileName: string, fstat: IFsStats) => fileName !== rootNodeModules);
+			let typeScriptFiles = _.filter(projectFiles, file => path.extname(file) === ".ts");
+			let definitionFiles = _.filter(typeScriptFiles, file => _.endsWith(file, ".d.ts"));
+			return { definitionFiles: definitionFiles, typeScriptFiles: typeScriptFiles };
+		}).future<ITypeScriptFiles>()();
+	}
+
+	public isTypeScriptProject(projectDir: string): IFuture<boolean> {
+		return ((): boolean => {
+			let typeScriptFiles = this.getTypeScriptFiles(projectDir).wait();
+
+			if (typeScriptFiles.typeScriptFiles.length > typeScriptFiles.definitionFiles.length) { // We need this check because some of non-typescript templates(for example KendoUI.Strip) contain typescript definition files
+				return true;
+			}
+
+			return false;
+		}).future<boolean>()();
+	}
+
+	private hasTsConfigFile(projectDir: string): IFuture<boolean> {
+		return this.$fs.exists(this.getPathToTsConfigFile(projectDir));
+	}
+
+	private getPathToTsConfigFile(projectDir: string): string {
+		return path.join(projectDir, "tsconfig.json");
```

Comment:
You could place the `"tsconfig.json"` string in a constant and use it from there. As it is not used anywhere else this is a minor comment and you can feel free to disregard it